src/history/taskfailedevent.ts

```diff
@@ -16,6 +16,18 @@ export class TaskFailedEvent extends HistoryEvent {
             options.timestamp,
         );
 
+        if (options.taskScheduledId === undefined) {
+            throw new Error("TaskFailedEvent needs a task scheduled id provided.");
+        }
+
+        if (options.reason === undefined) {
+            throw new Error("TaskFailedEvent needs a reason provided.");
+        }
+
+        if (options.details === undefined) {
+            throw new Error("TaskFailedEvent needs details provided.");
```

Comment:
I wonder if it's really worth throwing if `reason` or `details` are undefined. As far as I can tell, these are not required to be populated in DurableTask.Core.